### ISSUE TITLE
New ItemTag class isn't serialized/deserialized correctly

### ISSUE THREAD

**Issue by diogotcorreia** [SpigotMC/BungeeCord#2910]
This class: https://github.com/SpigotMC/BungeeCord/blob/master/chat/src/main/java/net/md_5/bungee/api/chat/ItemTag.java

It seems from the implementation that all fields except for `nbt` are completely useless. Even if you use the `ItemTag.Builder` class to construct the JSON, it isn't parsed back into JSON correctly.
Also, all the other fields are private and don't have any getters, which I guess they should if it's supposed to be an API.

TL;DR. All fields are null except for `nbt`.

**Comment by Mystiflow** [comment]
I wrote these fields in an initial update, and removed it as I didn't have time to fix the serialisation, I left the tags there to remind myself but they probably should be commented out instead. Side node; how did you make the builder?- it is private

**Comment by diogotcorreia** [comment]
@Mystiflow Alright makes sense. Maybe you or someone else can implement proper serialization? That'd help a lot. I can handle not having it, but it'd make more sense to support it as well.

Yeah, I just realised that afterwards. I was looking at the decompiled source code and not the source on GitHub, so the `@Builder` lombok tag was expanded. I haven't tried using it tho.

**Comment by Mystiflow** [comment]
I've [been working on it](https://github.com/Mystiflow/BungeeCord/tree/itemtag-serialising). Only thing missing is [this part](https://github.com/Mystiflow/BungeeCord/blob/itemtag-serialising/chat/src/main/java/net/md_5/bungee/api/chat/ItemTag.java#L98).

**Comment by md-5** [comment]
https://github.com/SpigotMC/BungeeCord/commit/c1522ab94c9e0866d660a151ed81ee2fa3280140
